Ship article website 1345

Ali6461 opened the pull request.

Ali6461 committed:
preparing for version patch and publish

Ali6461 committed:
0.7.90

Ali6461 committed:
0.7.91

Ali6461 committed:
preparing for version patch and publish

Ali6461 committed:
0.7.90

Ali6461 committed:
0.7.91

Ali6461 committed:
0.7.92

Ali6461 committed:
preparing for version patch and publish

Ali6461 committed:
0.7.93

Ali6461 committed:
0.7.94

Ali6461 committed:
Removed bundle-website from Fluid.purs and added it to the bin entry

Ali6461 committed:
0.7.95

Ali6461 committed:
Track article site in branch

Ali6461 committed:
Removed bundle-website from Fluid.purs and added it to the bin entry

Ali6461 committed:
Removed bundle-website from Fluid.purs and added it to the bin entry

Ali6461 committed:
Removed all remenants of bundle-website from Fluid.purs and added it to bin entry

rolyp commented on website/article/convolution/index.html:
```diff
@@ -6,10 +6,17 @@
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Fluid: Data-Linked Visualisations</title>
+<<<<<<< HEAD
```
It looks like something got accidentally committed before the merge conflict was resolved. You need to delete either the "Current change" or the "Incoming change". I think you want the versions with the `"../" prefixes on the URLs.

rolyp commented on package.json:
```diff
@@ -23,7 +23,9 @@
     "script/bundle-page.sh",
     "script/util/clean.sh",
-    "script/util/lisp-case.sh"
+    "script/util/lisp-case.sh",
```
It looks like we need `lisp-case.sh` because it's used by `test-page.sh`. I think this is actually a hangover from an old approach where website names were written in `CamelCase` and had to be converted to `lisp-case` by a script. That convention has gone, so I think we can remove the use of `lisp-case.sh` from `test-page.sh` and just use the value of `WEBSITE` unmodified. Then we can remove `lisp-case.sh` from here and indeed from `script/util` in the Fluid repo, because it will no longer be used anywhere.

rolyp commented on website/article/convolution/index.html:
```diff
@@ -82,7 +89,11 @@ <h4>Matrix convolution</h4>
 </div>
 
 <script type="module">
+<<<<<<< HEAD
```
Same situation here

rolyp commented on website/article/fluid/convolution.fld:
```diff
@@ -14,7 +14,10 @@ let convolve image kernel =
         (half_i, half_j) = (i `quot` 2, j `quot` 2);
         area = i * j
     in [| let interMatrix =
+<<<<<<< HEAD
```
Here you want the version with the comment

rolyp commented on website/article/index.html:
```diff
@@ -4,7 +4,11 @@
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
```
Here you want `Fluid Article Examples` as the page tilte

Ali6461 replied:
Oh okay, ive fixed them all now sorry about that, still getting use to it haha

rolyp reviewed (changes requested):
Hi Ali, looking much better. There are a couple of places where it looks like you did a `commit` before resolving all conflicts. If you get merge conflict, VSCode should show you where the conflicts are, and you need to make sure they're all resolved before committing. Usually tests will pick up any problems, but in this case it looks like we're not testing the `article` website as part of the automatic build.

Ali6461 committed:
Removed all remenants of bundle-website from Fluid.purs and added it to bin entry

Ali6461 committed:
picks up JS tests and added yarn run v1.22.22
$ ./script/bundle-website.sh article
Cleaning dist/article
Processing other static files:
Processing Fluid source files:
Processing shared JavaScript files:
Bundled website article
Done in 0.17s. to

Ali6461 committed:
0.7.96

Ali6461 committed:
picks up JS tests and added yarn run v1.22.22
$ ./script/bundle-website.sh article
Cleaning dist/article
Processing other static files:
Processing Fluid source files:
Processing shared JavaScript files:
Bundled website article
Done in 0.10s. to  and version bump

rolyp commented on script/test-all.sh:
```diff
@@ -3,3 +3,4 @@ set -xe
 
 . script/test.sh
 . script/test-website-all.sh
+./dist/fluid/shared/website-test.js article
```
I think we no longer need this, correct?

Ali6461 replied:
yes, its now redundant since itll already do that, let me remove it.

rolyp reviewed (changes requested):
Looking very good now Ali just one thing to fix I think, looks like there is an extra call to `website-test.js article` in `test-all.sh`.

Ali6461 committed:
picks up JS tests and added yarn run v1.22.22
$ ./script/bundle-website.sh article
Cleaning dist/article
Processing other static files:
Processing Fluid source files:
Processing shared JavaScript files:
Bundled website article
Done in 0.14s. to  and version bump

rolyp commented on script/install-article.sh:
```diff
@@ -0,0 +1,6 @@
+
+rm -rf website
+mkdir -p website
+cp -r node_modules/@explorable-viz/fluid/website/article website
```
I won't stop this from merging the PR as we can experiment with this when we get to `fluid-article`, but I think we'll want to change this so that it fails if `article` already exists in the target directory. (To force the user to delete it first, if that's indeed what they want to do.)

rolyp commented:
@Ali6461 Ok I've approved the PR, just a minor conflict to resolve and we're good to go!

Ali6461 commented:
> @Ali6461 Ok I've approved the PR, just a minor conflict to resolve and we're good to go!

im not sure about that conflict since there is nothing to change? when I do git commit there is no files changed either. Do i just re-push it or do i have to merge?

Ali6461 committed:
Merge branch 'develop' into ship-article-website-1345
